Sources/StreamChat/ChatClient.swift

```diff
@@ -410,7 +410,17 @@ extension ClientError {
 extension _ChatClient: ConnectionStateDelegate {
     func webSocketClient(_ client: WebSocketClient, didUpdateConectionState state: WebSocketConnectionState) {
         connectionStatus = .init(webSocketConnectionState: state)
-
+        
+        // Reacquire a new token if current one has expired
+        if case let .disconnected(error) = state,
```

Comment:
@b-onc I think this will actually be handled automatically in most cases. When the token expires, the socket is disconnected, and API calls wait for the new connection id. I guess it's safe to assume that we have the majority of the cases here covered. cc @dmigach